### PR TITLE
fix(controller): disable strict mode for mpi examples

### PR DESCRIPTION
**Pull request by shinytang6**
Can not find a better to solve [this issue](https://github.com/volcano-sh/volcano/pull/1263#issuecomment-766567292)
Signed-off-by: shinytang6 <1074461480@qq.com>

### CODE REVIEW

File: example/integrations/mpi/mpi-example.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ spec:
                 - /bin/sh
                 - -c
                 - |
+                  cat /etc/ssh/sshd_config | grep -v StrictModes > /etc/ssh/sshd_config.new && \
```
Comment (k82cn) [inline example/integrations/mpi/mpi-example.yaml:24]: Why do we need this one? It's a little complex to me :)

**Comment by shinytang6** [comment]
@k82cn After I removed the `subpath` in [PR 1263](https://github.com/volcano-sh/volcano/pull/1263), secret volume relies on symlinks to update secret, ssh secret directory looks something like this:
```
drwxr-xr-x 2 root root 120 Jan 28 06:43 ..2021_01_28_06_43_07.443659693/
lrwxrwxrwx 1 root root  31 Jan 28 06:43 ..data -> ..2021_01_28_06_43_07.443659693/
lrwxrwxrwx 1 root root  22 Jan 25 03:54 authorized_keys -> ..data/authorized_keys
lrwxrwxrwx 1 root root  13 Jan 25 03:54 config -> ..data/config
lrwxrwxrwx 1 root root  13 Jan 25 03:54 id_rsa -> ..data/id_rsa
lrwxrwxrwx 1 root root  17 Jan 25 03:54 id_rsa.pub -> ..data/id_rsa.pub
```
which will cause ssh communication between pods to fail due to directory permissions.

lf use `subpath` as before, these symlinks will not be generated(so that ssh communication works fine), but the flaw is that secret updates cannot be received.

The only way l thought of was to modify the server `sshd_config` to avoid that..

**Comment by k82cn** [comment]
> lf use subpath as before, these symlinks will not be generated(so that ssh communication works fine), but the flaw is that secret updates cannot be received.

Is it possible to enhance k8s for that case?

**Comment by shinytang6** [comment]
> > lf use subpath as before, these symlinks will not be generated(so that ssh communication works fine), but the flaw is that secret updates cannot be received.
> 
> Is it possible to enhance k8s for that case?

The k8s official does not seem to have this plan recently, the documentation has clearly pointed out this problem.
```
Note: A container using a Secret as a subPath volume mount will not receive Secret updates.
```

There are also some discussions here: [https://github.com/kubernetes/kubernetes/issues/50345](https://github.com/kubernetes/kubernetes/issues/50345)

**Comment by k82cn** [comment]
hm… interesting, I think we can share our case to upstream for more input.

**Comment by shinytang6** [comment]
> hm… interesting, I think we can share our case to upstream for more input.

Okay, I will raise an issue to k8s later. But WDYT is a temporary solution by now, to revert subpath changes or modify the sshd configuration? Due to the reason that the current codebase will make mpi-job unavailable.

**Comment by k82cn** [comment]
prefer to revert subpath changes :)

**Comment by Thor-wl** [comment]
I'm ok about the commits. Please resolve the conflicts.

**Pull request closed without merging**